Update index.html

EyeMagic opened the pull request:
Added <audio> elements for each of the buttons, as currently configured without the "Bank" switch as "FCC Drum Machine" example shows. I then added a simple <script> that uses a 'playback' class.

Jun10th2019: Planning to add `onclick='document.getElementById("display").innerHTML =` 'display' functionality unless there are suggestions for more efficiency???

EyeMagic committed:
Update index.html

Added <audio> elements for each of the buttons, as currently configured without the "Bank" switch as "FCC Drum Machine" example shows. I then added a simple <script> that uses a 'playback' class.

EyeMagic committed:
Controls added: Could use some styling changes and suggestions on options for keypress actions, etc.

jessicadavey reviewed (approved):
Hey, wow, it looks really good!

Here's a suggestion: The following code has been repeated for every button:
`onclick='document.getElementById("display").innerHTML = "[name of audio clip]'`

I think there's another way to do this.  Here's what I came up with:
```

  const buttons = [...document.querySelectorAll("buttons")];
    buttons.map(button => button.addEventListener("click", function (e) {

      const nameOfClip = ($(this).next('audio').get(0).id);
      const display = document.getElementById("display");
      display.innerHTML = nameOfClip;

    }));

```

First I created an array of all the button elements (using ES6 spread notation) and assigned it to a constant called "buttons". (If there end up being other buttons on the app, we could use ".playback" to select anything that had a class of playback, or something like that.)

I then used 'map' (a higher-order function, also an ES6 thing I think) to assign an event listener to each of the button elements in the buttons array.  This tells each button to listen for a "click", and when that happens, to execute the function that comes next.

The first line of the function uses jQuery to get the id of the audio element that goes with that button. (There may be a sexier way to do this; I'm by no means a jQuery expert.) Lucky for us, this happens to be the name of the clip itself (I'm sure eventually we'll change these to be actual drum sound names).  This is assigned to the constant "nameOfClip", and then the final two lines set this string to the display element.

Phew.  Anyway, I'm going to click "approve" because I'd be happy to accept the changes you've made and refactor it later.  Good job on taking the initiative and getting it done!  Let's see what @MellowMilieu has to say.  :)

EyeMagic commented:
Hey @jessicadavey Re: #8, thx for the excellent const code. I incorporated it and eliminated each 'display' call inside of the <button> element. Also, I moved the js code to an external file then cleaned up the id names as suggested by @MellowMilieu. I'm not quite sure where we're going direction wise, and your .playback suggestion sounds great, so I left that 'play' functionality as is for now. I'm assuming later when additional 'bank' or other switching (i.e 'Power' & 'Bank') decisions are made I'll have a better understanding what other solutions/logic are suggested and/or available. So, I left the 'audio' element outside of the 'button' element for now and think I'll leave that coding judgement for better js experts 🤔. I tagged you in the 'audio.js' file... hope this newbie did that correctly 😳. Please let me know if I missed anything or if anyone has any other great comments/suggestions.  👏

EyeMagic committed:
incorporated PR suggestions from Jun12th2019

Commit:
Added jquery intellisense for vs code

ghost commented:
Hey guys, good work! A quick reiteration of what I put on the Discord:

In the JS file we don't need to bind on-click events twice.  `$(".playback").click(function (e) { ... }` binds on-click to the buttons and so does `buttons.map(button => button.addEventListener("click", function (e) { ... }`. Instead, we can define a listener once for when buttons are clicked, and then perform any number of actions in the callback function `function (e) { ... }`. You're also using both a jQuery selector (`$(".playback")`) and an ES6 selector (`document.querySelectorAll("buttons")`), so I'd recommend picking one and sticking with it. This code, when placed in the audio.js file, should accomplish the same thing:
```
const display = document.getElementById("display");

$(".playback").click(function (e) {
    e.preventDefault();

    var clip = $(this).next('audio').get(0);

    display.innerHTML = clip.id;

    // TODO: refactor to account for rapid clicks
    if (clip.paused)
        clip.play();
    else
        clip.pause();
});
```

I added a JSON file a moment ago which simply adds jQuery intellisense to VS Code. Let me know if it doesn't work.

I haven't reviewed the HTML yet.

Commit:
Minor stylistic edits and updated JS audio functionality for repeated clicks

Commit:
Merge pull request #9 from nashvillefcc/index-audio-edits

Minor edits
